src/Components/Shared/src/WebEventData.cs

```diff
@@ -3,57 +3,36 @@
 
 using System;
 using System.Text.Json;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Components.Rendering;
-using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.Web
 {
-    /// <summary>
-    /// Provides mechanisms for dispatching events to components in a <see cref="Renderer"/>.
-    /// </summary>
-    public static class RendererRegistryEventDispatcher
+    internal class WebEventData
     {
-        /// <summary>
-        /// For framework use only.
-        /// </summary>
-        [JSInvokable(nameof(DispatchEvent))]
-        public static Task DispatchEvent(BrowserEventDescriptor eventDescriptor, string eventArgsJson)
+        // This class represents the second half of parsing incoming event data,
+        // once the type of the eventArgs becomes known.
+
+        public WebEventData(string eventDescriptorJson, string eventArgsJson)
+            : this(Deserialize<WebEventDescriptor>(eventDescriptorJson), eventArgsJson)
```

Comment:
This kind of machination is the reason why we usually avoid doing things that might fail in constructors. Consider just making this a static `Parse` method. This is just a nitpick since this is non-public.

Reply:
OK, done.